lib/game.rb

```diff
@@ -0,0 +1,71 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2, :token
+
+  WIN_COMBINATIONS = [
+  [0,1,2], #Top row
+  [3,4,5], #Middle row
+  [6,7,8], #Bottom row
+  [0,3,6], #Left column
+  [1,4,7], #Middle column
+  [2,5,8], #Right column
+  [0,4,8], #Left start diagonal
+  [2,4,6]  #Right start diagonal
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    if board.turn_count % 2 == 0
+      player_1
+    elsif board.turn_count % 2 == 1
+      player_2
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combination|
+      board.cells[combination[0]] == board.cells[combination[1]] &&
+      board.cells[combination[0]] == board.cells[combination[2]] && board.cells[combination[0]] != " "
+    end
+  end
+
+  def draw?
+    board.full? && !won?
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def winner
+    if won?
+      board.cells[won?[0]]
+    end
+  end
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.